preload.ts

```diff
@@ -1,23 +1,30 @@
 import { HyperImportConfig } from "./src/types";
 import { debugLog } from "./src/utils";
 
-const cwd = process.cwd();
-const config: HyperImportConfig = (await import(`${cwd}/bunfig.toml`)).default.hyperimport;
+const cwd = process.cwd(),
+    config: HyperImportConfig = (await import(`${cwd}/bunfig.toml`)).default.hyperimport;
 
 debugLog(config.debug, 3, "registering loaders...");
 
-for (const loader of config.loaders ?? []) {
-    await import(`./src/loaders/${loader}.ts`).then(async l => {
-        const plugin = await new l.default(cwd).toPlugin();
+for (const loader of config.loaders ?? []) 
+    try {
```

Comment:
Adding a catch function directly when awaiting the import makes it easier to read and is concise too. I would recommend to revert this change.